map/style_tests/style_symbols_consistency_test.cpp

```diff
@@ -61,6 +62,58 @@ set<string> GetSymbolsSetFromResourcesFile(string const & density)
   return symbols;
 }
 
+template <typename TIterator, typename TFn>
+string Join(TIterator begin, TIterator end, TFn && fn)
```

Comment:
I think JoinStrings can be improved. 
Now despite it 'joins strings', template allows to accumulate  any container values, for example:
vector<int> v = {1,2,3};
JoinStrings(v.begin(), v.end(), 1); also would be compiled and returned 8. 
It is funny, but it is ok.
to join strings transformed iterator MUST return strings, that means, that transformation functor must be responsible for converting any type to string. And delimiter must be string.
This requirement can be relaxed if rewrite code with using ostringstream:
template <typename TIterator, typename TDelimiter>
typename string JoinStrings(TIterator begin, TIterator end, TDelimiter const & delimiter)
{
  if (begin == end)
    return string();
  ostringstream o << *begin++;
  for (TIterator it = begin; it != end; ++it)
  {
    o << delimiter << *it;
  }
  return o.str();
}
Then transfomation function must transform to any supported by '<<' type, or in string.
what do you think?